modules/fluid_properties/src/fluidproperties/SalineMoltenSaltFluidProperties.C

```diff
@@ -0,0 +1,149 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+//
+#include "SalineMoltenSaltFluidProperties.h"
+
+registerMooseObject("FluidPropertiesApp", SalineMoltenSaltFluidProperties);
+
+InputParameters
+SalineMoltenSaltFluidProperties::validParams()
+{
+  InputParameters params = SinglePhaseFluidProperties::validParams();
+  params.addClassDescription("Molten salt fluid properties using Saline");
+  params.addRequiredParam<std::string>(
+      "comp_name", "The name of the components in the salt (e.g., 'LiF-NaF-KF')");
+  params.addRequiredParam<std::string>(
+      "comp_val", "The mole fraction of each salt component (e.g., '0.5-0.1-0.4)");
+  params.addRequiredParam<std::string>(
+      "prop_def",
+      "Definition of a fluid property file, which must be a file path to the "
+      "comma-separated data matching the Saline format.");
+  return params;
+}
+
+SalineMoltenSaltFluidProperties::SalineMoltenSaltFluidProperties(const InputParameters & parameters)
+  : SinglePhaseFluidProperties(parameters)
+{
+
+#ifdef SALINE_ENABLED
+
+  std::string propDef = getParam<std::string>("prop_def");
+  _d.load(propDef);
+  bool success = _tp.initialize(&_d);
+  if (!success)
+    mooseError("The initialize of the Saline interface has failed");
+  std::string name = getParam<std::string>("comp_name");
+  std::string comp = getParam<std::string>("comp_val");
+  std::vector<std::string> nameList;
+  MooseUtils::tokenize<std::string>(name, nameList, 1, "-");
+  std::vector<Real> valList;
+  MooseUtils::tokenizeAndConvert<Real>(comp, valList, "-");
+  Real mole_sum = 0.0;
+  for (Real val : valList)
+    mole_sum = mole_sum + val;
+  if (std::abs(mole_sum - 1.0) > 1e-6)
+    mooseError("Mole fractions of defined salt compound do not sum to 1.0.");
+  success = _tp.setComposition(nameList, valList);
+  if (!success)
+    mooseError("The composition set has failed");
+  _fluid_name = name;
+
+#endif
+#ifndef SALINE_ENABLED
+
+  mooseError("Saline was not made available during the build and can not be used. Make sure you "
+             "have the contrib/saline submodule checked out.");
+
+#endif
+}
+
+// Unit conversion constants for communicating with Saline
+const Real kPa_to_Pa = 1.0e3;
+const Real Pa_to_kPa = 1.0 / kPa_to_Pa;
+const Real kg_to_g = 1.0e3;
+const Real g_to_kg = 1.0 / kg_to_g;
+const Real m_to_cm = 1.0e2;
+const Real cm_to_m = 1.0 / m_to_cm;
+const Real N_to_mN = 1.0e3;
+const Real mN_to_N = 1 / N_to_mN;
```

Comment:
I think these should either be `constexpr` or `static` class data members. At a minimum I would put these in an anonymous namespace but even that can run into collisions because of unity builds

Reply:
You want to declare these as `constexpr` and also put them all into an anonymous namespace in this file?

Reply:
Or make them class data members